### PR TITLE
SyntheSimJava Documentation

### PR DESCRIPTION
**Pull request by azaleacolburn**
### Description
Documentation for SyntheSimJava

Let me know if anything else should be added.

No JIRA Issue

### CODE REVIEW

**Review by BrandonPacewic — APPROVED** [review]
⭐

**Review by a-crowell — APPROVED** [review]
LGTM!

File: simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/revrobotics/CANSparkMax.java [hunk 1]
```diff
@@ -55,16 +62,24 @@ public REVLibError setIdleMode(com.revrobotics.CANSparkBase.IdleMode mode) {
...
-    public void newFollower(CANSparkMax f) {
+    void newFollower(CANSparkMax f) {
         this.followers.add(f);
     }
 
-    // Must pass in a simulation-supported leader to have the simulated portion of this motor follow
+    /* 
+     * Causes a simulation-supported leader to follow another simulation-supported leader
+     * The real versions of these motors will also follow each other
+     *
```
Comment (PepperLola) [inline simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/revrobotics/CANSparkMax.java:102]:
```suggestion
```

File: simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/revrobotics/SparkAbsoluteEncoder.java [hunk 2]
```diff
@@ -9,59 +9,102 @@ public class SparkAbsoluteEncoder implements AbsoluteEncoder {
...
+     * All other parameters are returned from the real motor, which likely won't exist, not sure what it does then but we'll just call it UB
+     */
     public SparkAbsoluteEncoder(com.revrobotics.SparkAbsoluteEncoder realEncoder, CANEncoder simEncoder) {
         this.realEncoder = realEncoder;
         this.simEncoder = simEncoder;
     }
 
+    /*
+     * Get the average sampling depth for an absolute encoder
+     * Returns: The average sampling depth
```
Comment (PepperLola) [inline simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/revrobotics/SparkAbsoluteEncoder.java]:
```suggestion
     * @return The average sampling depth
```

**Review by PepperLola — CHANGES_REQUESTED** [review]
I think it's worth changing the Java comments to use the [Javadoc](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html) format so people who use the library can get intellisense native to their editor.

File: simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/ctre/TalonFX.java [hunk 3]
```diff
@@ -22,20 +22,36 @@ public TalonFX(int deviceNumber) {
         this.m_motor = new CANMotor("SYN TalonFX", deviceNumber, 0.0, false, 0.3);
     }
 
-    /// I think we're getting percentOutput and speed mixed up
+    /// I think we're getting percentOutput and torque mixed up
```
Comment (HunterBarclay) [inline simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/ctre/TalonFX.java]: Still keeping this comment here?

Comment (azaleacolburn) [reply simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/ctre/TalonFX.java]: oh ya thanks

Comment (azaleacolburn) [reply simulation/SyntheSimJava/src/main/java/com/autodesk/synthesis/ctre/TalonFX.java]:
```suggestion
```

**Review by PepperLola — APPROVED** [review]
I think it looks good now. I made some changes that you're free to revert if you want.

Edit: It looks like I introduced a merge conflict by adding a class comment. I think that should stay there but maybe the import should be removed if it was added in this PR?

**Review by BrandonPacewic — COMMENTED** [review]
> I think it looks good now. I made some changes that you're free to revert if you want.
> 
> Edit: It looks like I introduced a merge conflict by adding a class comment. I think that should stay there but maybe the import should be removed if it was added in this PR?

The merge conflict was just due to this branch not being up to date with dev where that commented import was already removed.

Looks good to go as long as @azaleacolburn is ok with the pushed changes ✅

**Pull request merged**